src/Arcus.Observability.Telemetry.Serilog.Filters/TelemetryTypeFilter.cs

```diff
@@ -44,27 +48,36 @@ public static TelemetryTypeFilter On(TelemetryType telemetryType)
         /// </summary>
         /// <param name="telemetryType">Current telemetry type that should be filtered</param>
         /// <param name="isTrackingEnabled">Indication whether or not this telemetry type should be tracked</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="telemetryType"/> is outside the bounds of the enumeration.</exception>
         public static TelemetryTypeFilter On(TelemetryType telemetryType, bool isTrackingEnabled)
         {
+            Guard.For(() => !Enum.IsDefined(typeof(TelemetryType), telemetryType), 
+                new ArgumentOutOfRangeException(nameof(telemetryType), telemetryType, "Requires a type of telemetry that's within the supported value range of the enumeration"));
+            
             return new TelemetryTypeFilter(telemetryType, isTrackingEnabled);
         }
 
         /// <inheritdoc />
         public bool IsEnabled(LogEvent logEvent)
         {
-            if (IsTrackingEnabled == true)
+            if (IsTrackingEnabled is true)
             {
                 return true;
             }
 
-            var telemetryType = logEvent.Properties.GetAsEnum<TelemetryType>(ContextProperties.General.TelemetryType);
-            if (telemetryType == null)
+            switch (TelemetryType)
             {
-                return true;
+                case TelemetryType.Dependency:
+                    return logEvent.Properties.ContainsKey(ContextProperties.DependencyTracking.DependencyLogEntry);
+                case TelemetryType.Request:
+                    return logEvent.Properties.ContainsKey(ContextProperties.RequestTracking.RequestLogEntry);
+                case TelemetryType.Events:
+                    return logEvent.Properties.ContainsKey(ContextProperties.EventTracking.EventLogEntry);
+                case TelemetryType.Metrics:
+                    return logEvent.Properties.ContainsKey(ContextProperties.MetricTracking.MetricLogEntry);
+                default:
+                    return false;
```

Comment:
Problem here: the `TelemetryType` also includes `Trace`, but since we don't do traces ourselves, everything outside these cases is considered being a trace. Not sure if I should remove the enum value or add a case here and always return `false`.

Reply:
When is this method called ?  I would assume that it's called to determine if the event must be logged or not ?  Isnt that done based on the configured loglevel ?  Or am I talking rubbish here ?

Reply:
> When is this method called ?

Serilog filters are predicates on the log level. They're called before the the event is logged to the sink.

> I would assume that it's called to determine if the event must be logged or not ?

Yes.

>  Isnt that done based on the configured loglevel ?

Yes, also, but a filter can be added for extra more grainer control over the events being logged.

> Or am I talking rubbish here ?

Not at all 😉 .